src/functions/index.ts

```diff
@@ -74,6 +75,12 @@ export class FunctionRegistry extends Registry<FunctionDescription> {
   } = {};
 
   add(name: string, addDescr: AddFunctionDescription) {
+    name = name.toUpperCase();
+    // @ts-ignore
+    return super.add(name, addDescr);
```

Comment:
here also, why does the `add` has all the logic removed ? and a brand new ts-ignore ?

Reply:
But super.add() will call replace, so there is no need for duplicating the logic ?

@ts-ignore is because this registry add take as argument something different from the object that will be stored, it allows to use the generic registry as a base which doesn't support this usecase (type-wise only)

Reply:
Oh I didn't catch that, indeed. 

nitpick: I think I'd prefer to not have a ts-ignore, do the check of key duplication here, and call `this.replace` instead of `this.add`.